packages/server/index.js

```diff
@@ -1,11 +1,185 @@
+                'Content-Type': 'application/x-www-form-urlencoded' 
+            },
+        };
+
+        const sessionDetails  = await axios.post(authOptions.url, querystring.stringify(authOptions.form), {"headers": authOptions.headers}).catch(r => console.error(r));
+        const userData = await getUserProfile(sessionDetails.data) // .id
+        const topTracks = await getUserTopTracks(sessionDetails.data)
+        const trackIds = topTracks.map(b => b.id).join(',')
+        const audioFeatures = await getAudioFeatures(sessionDetails.data, trackIds)
+
+
+        res.send(
+            {
+                userData,
+                topTracks, 
+                session: sessionDetails.data
+            })
+        // Insert data into DB after returning tracks. Reduces load time for user
+        for (let i = 0; i < audioFeatures.length; i++){
+            console.log(i)
+            const mergeTrack = {...audioFeatures[i], ...topTracks[i] }
+
+            const ranking = i + 1
+
+            const query = `MERGE (u:user {name: "${userData.id}"})
+                            MERGE (s:song {name: "${mergeTrack.name.replace(/"/g, '\'')}", acousticness: ${mergeTrack.acousticness}, danceability: ${mergeTrack.danceability}, energy: ${mergeTrack.energy}, tempo: ${mergeTrack.tempo}, valence: ${mergeTrack.valence}})
+                            MERGE (u)-[:LISTENS_TO {ranking: ${ranking}}]-(s)
+                            RETURN u, s`
+            
+            await session.writeTransaction(tx => tx.run (query, { userData, mergeTrack, ranking }))
+        }
+    }
+})
+
+app.get('/err', (req, res) => {
+    res.status(400).send('Something broke!')
+});
+
+app.get('/login', (req, res) => {
+    const state = crypto.randomBytes(20).toString('hex');
+    const scope = 'user-read-private user-read-email user-top-read';
+
+    res.redirect('https://accounts.spotify.com/authorize?' +
+        querystring.stringify({
+            response_type: 'code',
+            client_id: process.env.SPOTIFY_CLIENT_ID,
+            scope: scope,
+            redirect_uri: redirect_uri,
+            state: state,
+            show_dialog: true,
+            code_challenge_method: 'S256',
+            code_challenge: code_challenge
+        }));
 })
 
 app.listen(port, () => {
     console.log(`Example app listening on port ${port}`)
-})
\ No newline at end of file
+})
+
+app.get('/blend', (req,res) => {
+    const currUser = req.params.currUser
+    const targetUser = req.params.targetUser
+    // neo4j query here. Both of these parameters are spotify usernames
+    // TODO: replace instance of 'daflyingcactus' and 'emilydinh' with currUser and targetUser respectively
+    // this query current compares by average energy. we can just copy and paste this and switch out for other audio properties (valence, danceability,...)
+    const query = `match (u:user)--(s:song)
```

Comment:
i mentioned in the gc, but i think we could also include another param for which audio property to base the blend off of.
some options from the labels we have in the neo4j db could be: `energy`, `danceability`, `valence`